### PR TITLE
Add secret hooks

### PR DESCRIPTION
**Pull request by wallyworld**
The meta.Hooks() method was missing the secrets hooks. Add these.

Also update go.mod to address a CVE in `golang.org/x/net`.

### CODE REVIEW

File: hooks/hooks.go [hunk 1]
```diff
@@ -18,7 +18,6 @@ const (
 	UpgradeCharm          Kind = "upgrade-charm"
 	Stop                  Kind = "stop"
 	Remove                Kind = "remove"
-	Action                Kind = "action"
```
Comment (hpidcock) [inline hooks/hooks.go:21]: this is used by the uniter

Comment (hpidcock) [reply hooks/hooks.go:21]: are you going to remove the code there?

**Review by hpidcock — APPROVED** [review]

File: hooks/hooks.go (hunk 1 above)
Comment (wallyworld) [reply hooks/hooks.go:21]: ah, came up as unused for me when i searched, but i missed the usage, will add it back

**Pull request merged**